### PR TITLE
add yaml configure file support

### PR DESCRIPTION
**Pull request by fangdingjun**
add support for loading configure from yaml file

### CODE REVIEW

File: file.go [hunk 1]
```diff
@@ -79,6 +81,51 @@ func (j *JSONLoader) Load(s interface{}) error {
...
+// Load loads the source into the config defined by struct s
+// Defaults to using the Reader if provided, otherwise tries to read from the
+// file
+func (y *YAMLLoader) Load(s interface{}) error {
+	var data []byte
+	var err error
+
+	if y.Reader != nil {
+		data, err = ioutil.ReadAll(y.Reader)
+
```
Comment (rjeczalik) [inline file.go]: remove this line (excessive line)

**Comment by rjeczalik** [comment]
Hey @fangdingjun, thanks for your contribution!

File: file.go [hunk 2]
```diff
@@ -79,6 +81,51 @@ func (j *JSONLoader) Load(s interface{}) error {
...
+		var file *os.File
+
+		file, err = getConfig(y.Path)
+		if err != nil {
+			return err
+		}
+
+		data, err = ioutil.ReadAll(file)
+
+		file.Close()
```
Comment (cihangir) [inline file.go]: @rjeczalik what i remember is, it wasn't working properly with relative paths, so we added this. but i dont remember in which case it wasnt working. i've checked go release history but couldn't find anything regarding any fix might relate to this

**Comment by cihangir** [comment]
Hey @fangdingjun sorry for forgetting this issue, thanks for the contribution.

**Pull request merged**